cmd/agent/app/reporter/connect_metrics.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2020 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reporter
+
+import (
+	"github.com/uber/jaeger-lib/metrics"
+	"go.uber.org/zap"
+)
+
+type connectMetrics struct {
+	// used for reflect current connection stability
+	Reconnects metrics.Counter `metric:"collector_reconnects" help:"Number of successful connections (including reconnects) to the collector."`
+
+	// Connection status that jaeger-agent to jaeger-collector, 1 is connected, 0 is disconnected
+	Status metrics.Gauge `metric:"collector_connected" help:"Status of connection between the agent and the collector; 1 is connected, 0 is disconnected"`
+}
+
+// ConnectMetrics include connectMetrics necessary params if want to modify metrics of connectMetrics, must via ConnectMetrics API
+type ConnectMetrics struct {
+	Logger         *zap.Logger     // required
+	MetricsFactory metrics.Factory // required
+	connectMetrics *connectMetrics
+}
+
+// NewConnectMetrics will be initialize ConnectMetrics
+func (r *ConnectMetrics) NewConnectMetrics() {
```

Comment:
This is unusual construct. I would make it simply a constructor function
```go
func NewConnectMetrics(logger *zap.Logger, metrics metrics.Factory) *ConnectMetrics
```

Reply:
Ah, sorry for responding too late..

I saw that your PR https://github.com/jaegertracing/jaeger/pull/2728 and thanks for your help clean-up.

Looked forward to learning more from you